test/gtest/uct/test_peer_failure.cc

```diff
@@ -620,7 +625,8 @@ UCS_TEST_SKIP_COND_P(test_uct_peer_failure_keepalive, killed,
 
 UCS_TEST_SKIP_COND_P(test_uct_peer_failure_keepalive, killed_post_am,
                      !check_caps(UCT_IFACE_FLAG_EP_CHECK |
-                                 UCT_IFACE_FLAG_AM_SHORT))
+                                 UCT_IFACE_FLAG_AM_SHORT),
+                     "UD_TIMER_TICK?=2s")
```

Comment:
why needed?

Reply:
this is needed to start resending which would prevent us from doing AM operations (they return NO_RESOURCE)

Reply:
it seems weird, would the test fail without it?
and why do we expect m_err_count to be 2 ?

Reply:
> it seems weird, would the test fail without it?

yes, it will fail and wouldn’t test the original idea, because we need to do multiple AM sends without getting NO_RESOURCE

> and why do we expect m_err_count to be 2 ?

it is 2-second deadline

Reply:
> yes, it will fail and wouldn’t test the original idea, because we need to do multiple AM sends without getting NO_RESOURCE

Do you mean we need to avoid resending quickly?
What will be the failure?

> it is 2-second deadline

Ok

Can you please add a comment in the test to explain why need 2s timer tick?

Reply:
> > yes, it will fail and wouldn’t test the original idea, because we need to do multiple AM sends without getting NO_RESOURCE
> 
> Do you mean we need to avoid resending quickly? What will be the failure?

the failure will be returning `UCS_ERR_NO_RESOURCE` from `uct_ep_am_short`, but we want to test the case when AM operations are scheduled every `2` seconds to reproduce issue with setting `send_time` and not detecting an error


> > it is 2-second deadline
> 
> Ok
> 
> Can you please add a comment in the test to explain why need 2s timer tick?

done

Reply:
> the failure will be returning `UCS_ERR_NO_RESOURCE` from `uct_ep_am_short`, but we want to test the case when AM operations are scheduled every `2` seconds to reproduce issue with setting `send_time` and not detecting an error

Maybe we can test this in test_ud.cc instead?

Reply:
we don't have a test which continues sending when peer was killed, so it makes sense having it for all transports

Reply:
or you think that it worth moving it to move it to test_ud.cc, because it has dependency on UD behavior?

Reply:
unfortunately, we don't have peer failure tests in UD and we should introduce it first

Reply:
I'd say to remove this test for now. and then in next PR add UD specific test in test_ud.cc that includes packet drop by rx filter and making sure we start resending even if adding new packet every time.
in these tests we have better control over ud state.

Reply:
we don’t need resending, that’s why setting UD_TIMER_TICK to big enough value.
I’ll move this test to test_ud.cc